alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -53,50 +53,67 @@ void RunnableObject::start(RunningPlan* rp)
     _engine->editScheduler().schedule(std::bind(&RunnableObject::doInit, this));
 }
 
-void RunnableObject::setTerminatedState()
+bool RunnableObject::setTerminatedState()
 {
     ++_execState;
 
-    clearFlags(Flags::RUN_TRACED);
-    if (!areFlagsSet(Flags::INIT_EXECUTED)) {
+    _runTraced = false;
+    if (!_initExecuted) {
         _execContext.store(nullptr);
-        return;
+        return true;
     }
-    clearFlags(Flags::INIT_EXECUTED);
+    _initExecuted = false;
+    return false;
 }
```

Comment:
isn't the logic here wrong? if we didn't Initialize, we ` _execContext.store(nullptr);` and return true? I was expecting the method to return false if Init hadn't been called, and true otherwise. Please document this method

Reply:
I have documented this method, but as mentioned in the previous comment this code needs to be redesigned